contracts/extensions/MultisigPermissions.sol

```diff
@@ -0,0 +1,551 @@
+    globalThreshold = _globalThreshold;
+
+    emit GlobalThresholdSet(_globalThreshold);
+  }
+
+  function setDomainSkillThreshold(uint256 _domainSkillId, uint256 _threshold) public onlyCoreRoot {
+    domainSkillThreshold[_domainSkillId] = _threshold;
+
+    emit DomainSkillThresholdSet(_domainSkillId, _threshold);
+  }
+
+  function createMotion(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    address[] memory _targets,
+    bytes[] memory _data
+  ) public notDeprecated {
+    require(_data.length >= 1, "colony-multisig-invalid-motion");
+    require(_targets.length == _data.length, "colony-multisig-invalid-motion");
+
+    motionCount += 1;
+    Motion storage motion = motions[motionCount];
+    motion.targets = _targets;
+    motion.data = _data;
+    motion.creator = msgSender();
+    motion.creationTimestamp = block.timestamp;
+
+    for (uint256 i = 0; i < _data.length; i += 1) {
+      require(_targets[i] != address(colony), "multisig-passed-target-cannot-be-base-colony");
+
+      ActionSummary memory actionSummary = getActionSummary(
+        address(colonyNetwork),
+        address(colony),
+        _data[i],
+        _targets[i]
+      );
+
+      require(actionSummary.domainSkillId < type(uint256).max, "colony-multisig-invalid-motion");
+      require(
+        actionSummary.requiredPermissions != bytes32(type(uint256).max),
+        "colony-multisig-invalid-motion"
+      );
+
+      // slither-disable-next-line incorrect-equality
+      if (motion.domainSkillId == 0 && motion.requiredPermissions == 0) {
+        motion.domainSkillId = actionSummary.domainSkillId;
+        motion.requiredPermissions = actionSummary.requiredPermissions;
+      } else {
+        // slither-disable-next-line incorrect-equality
+        require(
+          motion.domainSkillId == actionSummary.domainSkillId &&
+            motion.requiredPermissions == actionSummary.requiredPermissions,
+          "colony-multisig-invalid-motion"
+        );
+      }
+    }
+
+    validateMotionDomain(_permissionDomainId, _childSkillIndex, motionCount);
+    validateUserPermissions(_permissionDomainId, motionCount);
+
+    emit MotionCreated(msgSender(), motionCount);
+
+    changeVoteFunctionality(_permissionDomainId, motionCount, Vote.Approve, true);
+  }
+
+  function changeVote(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _motionId,
+    Vote _vote
+  ) public motionExists(_motionId) notExecuted(_motionId) notRejected(_motionId) {
+    validateMotionDomain(_permissionDomainId, _childSkillIndex, _motionId);
+    changeVoteFunctionality(_permissionDomainId, _motionId, Vote.Approve, _vote == Vote.Approve);
+    changeVoteFunctionality(_permissionDomainId, _motionId, Vote.Reject, _vote == Vote.Reject);
+
+    if (_vote == Vote.Reject && checkThreshold(_motionId, Vote.Reject)) {
+      cancelFunctionality(_motionId);
+    }
+  }
+
+  function changeVoteFunctionality(
```

Comment:
Stylistic comment, but I wouldn't be opposed to the convention of prepending underscores to internal implementation functions, i.e. `_changeVote()` in lieu of `changeVoteFunctionality`. But up to you.

I also think we should move these private functions towards the bottom.

Reply:
Agree on the movement. I'm not 100% sold on the renaming; we have had slip-ups in terms of variables before that only differed by a prepended `_`. I think it's less likely to go unnoticed in this instance, though, as the result would probably be an infinite loop. I'll sit on it for now and see if I get more comfortable with the idea.